### ISSUE TITLE
SegFault when using NATS on WSL2

### ISSUE THREAD

**Issue by btmnk** [oven-sh/bun#11882]
### How can we reproduce the crash?

Nats v2.26.0
Bun v1.1.13

https://github.com/btmnk/test-bun-nats

It only segfaults on WSL2 with Ubuntu 20 and 22 (both LTS versions)

When I tried creating a repro on CodeSandbox it worked fine so I assume it has something to do with WSL.
I only got that error after upgrading to bun 1.1.13, it works fine when using 1.0.9.

### Relevant log output

```shell
============================================================
Bun v1.1.13 (bd6a6051) Linux x64
Args: "bun" "run" "index.ts"
Features: jsc transpiler_cache tsconfig
Builtins: "bun:main" "node:dns" "node:fs" "node:net" "node:path" "node:string_decoder" "node:tls" "node:util/types"
Elapsed: 48ms | User: 42ms | Sys: 10ms
RSS: 1.07GB | Peak: 77.41MB | Commit: 1.07GB | Faults: 0

panic(main thread): Segmentation fault at address 0x20
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.13/lr1bd6a605AiggggF2nk6kE+xyQ25i/jEA2AgC

[1]    692 illegal hardware instruction  bun run index.ts
```

### Stack Trace (bun.report)

Bun v1.1.13 ([`bd6a605`](<https://github.com/oven-sh/bun/tree/bd6a6051203ff577240615ff4e242270035a36b7>)) on linux x86_64 [RunCommand]

Segmentation fault at address 0x00000020

- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/bd6a6051203ff577240615ff4e242270035a36b7/ld-temp.o#L0>): `WTF::jscSignalHandler`
- `??`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/bd6a6051203ff577240615ff4e242270035a36b7/ld-temp.o#L0>): `end_hquery`

**Comment by puddleglum56** [comment]
Also having this same issue with v1.1.13 with NATS on WSL2. Downgrading Bun to v1.1.10 looks to have fixed it. Later versions have the seg fault.

**Comment by Jarred-Sumner** [comment]
I believe this is an issue in our usage of c-ares.

- `end_hquery`
- `next_lookup`
- `ares_getaddrinfo`
- [`c_ares.zig:580`](https://github.com/oven-sh/bun/blob/ff2080da1eea03939e52dc25248ce43a157318be/src/deps/c_ares.zig#L580): `src.bun.js.api.bun.dns_resolver.DNSResolver.doLookup`
- [`dns_resolver.zig:2500`](https://github.com/oven-sh/bun/blob/ff2080da1eea03939e52dc25248ce43a157318be/src/bun.js/api/bun/dns_resolver.zig#L2500): `Bun__DNSResolver__lookup`

**Comment by Jarred-Sumner** [comment]
This was fixed in #12223 and will be included in Bun v1.1.18

If you want to get it early, run `bun upgrade --canary`
